diff -uNrp linux-3.18.3-201.jk1.fc21.x86_64.org/sound/usb/quirks.c linux-3.18.3-201.jk1.fc21.x86_64.new/sound/usb/quirks.c
--- linux-3.18.3-201.jk1.fc21.x86_64.org/sound/usb/quirks.c	2015-01-26 18:37:29.328584333 +0100
+++ linux-3.18.3-201.jk1.fc21.x86_64.new/sound/usb/quirks.c	2015-01-26 18:38:52.352434730 +0100
@@ -1226,8 +1226,9 @@ u64 snd_usb_interface_dsd_format_quirks(
 
 	/* XMOS based USB DACs */
 	switch (chip->usb_id) {
-	/* iFi Audio micro/nano iDSD */
-	case USB_ID(0x20b1, 0x3008):
+	case USB_ID(0x20b1, 0x3008): /* iFi Audio micro/nano iDSD */
+	case USB_ID(0x20b1, 0x2008): /* Matrix Audio X-Sabre */
+	case USB_ID(0x20b1, 0x300a): /* Matrix Audio Mini-i Pro */
 		if (fp->altsetting == 2)
 			return SNDRV_PCM_FMTBIT_DSD_U32_BE;
 		break;
